### ISSUE TITLE
How to custom field Binauth

### ISSUE THREAD

**Issue by nongquanghuong** [openNDS/openNDS#33]
I want to add field to binauth, i find code and i see code at libmicrohttpd.c. I want add a new field as "data" to this, instead of default "username" and "password" and give it as param to binauth.sh.
I try edit code and build like script on "buildhttp.sh" on this git but it not work.

**Comment by bluewavenet** [comment]
@nongquanghuong 
This is implemented already in release 5.1.0, but is not in the documentation yet as far as I can see.
You can see how to use it from the installed library scripts and test it by enabling login option and binauth logging.
ie:
```
uci set opennds.@opennds[0].login_option_enabled='1'
uci set opennds.@opennds[0].binauth='/usr/lib/opennds/binauth_log.sh'
uci commit opennds
```
In /usr/lib/opennds/login.sh you will find:

```
	# You can also send a custom data string to BinAuth. Set the variable $custom to the desired value
	# Max length 256 characters
	custom="Custom data sent to BinAuth"

	echo "<br><italic-black> Your News or Advertising could be here, contact the owners of this Hotspot to find out how!</italic-black>"

	echo "<form action=\"/opennds_auth/\" method=\"get\">"
	echo "<input type=\"hidden\" name=\"tok\" value=\"$tok\">"
	echo "<input type=\"hidden\" name=\"redir\" value=\"$requested\"><br>"
	echo "<input type=\"hidden\" name=\"custom\" value=\"$custom\">"
	echo "<input type=\"submit\" value=\"Continue\" >"
	echo "</form><hr>"

```
Add this to the client dialogue by editing the line:
`	echo "<input type=\"text\" name=\"custom\" value=\"$custom\">"`

to read:

`	echo "<input type=\"text\" name=\"custom\" value=\"$custom\">"`

Finally activate the changes:

`service opennds restart`

**Comment by bluewavenet** [comment]
It is mentioned in the documentation but could do with some more detail.
https://opennds.readthedocs.io/en/stable/binauth.html?highlight=custom#example-2-local-nds-access-log

**Comment by nongquanghuong** [comment]
> @nongquanghuong
> This is implemented already in release 5.1.0, but is not in the documentation yet as far as I can see.
> You can see how to use it from the installed library scripts and test it by enabling login option and binauth logging.
> ie:
> 
> ```
> uci set opennds.@opennds[0].login_option_enabled='1'
> uci set opennds.@opennds[0].binauth='/usr/lib/opennds/binauth_log.sh'
> uci commit opennds
> ```
> 
> In /usr/lib/opennds/login.sh you will find:
> 
> ```
> 	# You can also send a custom data string to BinAuth. Set the variable $custom to the desired value
> 	# Max length 256 characters
> 	custom="Custom data sent to BinAuth"
> 
> 	echo "<br><italic-black> Your News or Advertising could be here, contact the owners of this Hotspot to find out how!</italic-black>"
> 
> 	echo "<form action=\"/opennds_auth/\" method=\"get\">"
> 	echo "<input type=\"hidden\" name=\"tok\" value=\"$tok\">"
> 	echo "<input type=\"hidden\" name=\"redir\" value=\"$requested\"><br>"
> 	echo "<input type=\"hidden\" name=\"custom\" value=\"$custom\">"
> 	echo "<input type=\"submit\" value=\"Continue\" >"
> 	echo "</form><hr>"
> ```
> 
> Add your custom string to the client dialogue by editing the line:
> ` echo "<input type=\"hidden\" name=\"custom\" value=\"$custom\">"`
> 
> to read:
> 
> ` echo "<input type=\"text\" name=\"custom\" value=\"$custom\">"`
> 
> Finally activate the changes:
> 
> `service opennds restart`
> 
> **_Edit:_**
> The log can be seen at /tmp/ndslog/binauthlog.log

thanks you.
